### PR TITLE
fix: removed ab_mock_mongo_schema_enabled, ab_gsheet_schema_enabled flag code

### LINKED ISSUE #28971: [Task]: Remove the code support for `ab_mock_mongo_schema_enabled` & `ab_gsheet_schema_enabled` flags
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- Since the `ab_mock_mongo_schema_enabled` & `ab_gsheet_schema_enabled` flags have been rolled to 100% of the users, then the code support should be removed.
- Add Cypress tests for the above code.

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description
Removed the code for `ab_mock_mongo_schema_enabled`, `ab_gsheet_schema_enabled` feature flags as they both have been rolled out to 100% of the users on Nov 20, 2023

#### PR fixes following issue(s)
Fixes #28971 
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Streamlined the process for generating CRUD pages from various data sources.
  - Enhanced the test suite for better coverage of data source functionalities.

- **Bug Fixes**
  - Fixed issues with the CRUD generation logic and improved the reliability of deployment mode verification.

- **Refactor**
  - Removed deprecated feature flags related to data source schema previews.
  - Simplified the logic for determining if a plugin can preview data.

- **Chores**
  - Updated Cypress tests to align with the latest CRUD generation methods and removed unnecessary feature flag setups.
  - Cleaned up test code for better maintainability and readability.

- **Documentation**
  - No visible documentation changes for end-users.

- **Style**
  - No visible style changes for end-users.

- **Tests**
  - Refined end-to-end tests to reflect changes in feature flag management.

- **Revert**
  - No reverts in this release.

### CODE REVIEW

**Review by hetunandu — APPROVED** [review]

File: app/client/cypress/e2e/Regression/ServerSide/GenerateCRUD/MySQL2_Spec.ts [hunk 1]
```diff
@@ -388,7 +388,7 @@ describe(
       () => {
         deployMode.DeployApp();
         deployMode.NavigateBacktoEditor();
-        dataSources.DeleteDatasourceFromWithinDS(dsName, 200);
+        dataSources.DeleteDatasourceFromWithinDS(dsName, 409);
```
Comment (sneha122) [inline app/client/cypress/e2e/Regression/ServerSide/GenerateCRUD/MySQL2_Spec.ts:391]: @AmanAgarwal041 Why was this 200 before? was it allowing deletion of datasources before?

File: app/client/cypress/e2e/Regression/ServerSide/GenerateCRUD/Postgres2_Spec.ts [hunk 2]
```diff
@@ -651,7 +651,7 @@ describe(
     it("18. Verify Deletion of the datasource when Pages/Actions associated are not removed yet", () => {
       deployMode.DeployApp();
       deployMode.NavigateBacktoEditor();
-      dataSources.DeleteDatasourceFromWithinDS(dsName, 200);
+      dataSources.DeleteDatasourceFromWithinDS(dsName, 409);
```
Comment (sneha122) [inline app/client/cypress/e2e/Regression/ServerSide/GenerateCRUD/Postgres2_Spec.ts:654]: @AmanAgarwal041 Same as above comment

**Comment by sneha122** [comment]
@AmanAgarwal041 Tested the PR with and without appsmith domains, I am able to see the generate CRUD functionality. Seems to be working as expected

File: app/client/cypress/e2e/Regression/ServerSide/GenerateCRUD/MySQL2_Spec.ts (hunk 1 above)
Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ServerSide/GenerateCRUD/MySQL2_Spec.ts:391]: I am not sure of it. Whether there was a change in the api or not. I tested out this spec and the other one, it wasn't able to delete the ds as there were actions using it. So, I changed the status code. @sneha122

**Review by sneha122 — APPROVED** [review]

**Pull request merged**